src/validation/3.0/invalid-property-value.rule.ts

```diff
@@ -138,6 +145,101 @@ export class Oas30InvalidPropertyValueValidationRule extends Oas30PathValidation
         return method === "put" || method === "post" || method === "options" || method === "patch";
     }
 
+    /**
+     * Returns resolved parameter definition or the address of the reference if parameter can not be resolved
+     * (e.g. when the pointer is to a non-existent value within the components).
+     * @param node - Parameter definition within a path item or operation
+     * @return {Oas30Parameter | string} - Returns (recursively) resolved parameter or a string with invalid ref.
+     */
+    private resolveParameter(node: Oas30Parameter): Oas30Parameter | string {
+        let resolvedParam = node;
+        while (resolvedParam.$ref !== undefined) {
+            const referencedNode = <Oas30Parameter>ReferenceUtil.resolveRef(resolvedParam.$ref, node.ownerDocument());
+            if (referencedNode === undefined || referencedNode === null) {
+                return resolvedParam.$ref;
+            }
+            resolvedParam = referencedNode;
+        }
+        return resolvedParam
+    }
+
+    /**
+     * Validates that all parameter name and "in" combinations are unique
+     * @param {Oas30Parameter[]} - List of parameters to check
+     * @return {boolean} - true if unique, false otherwise
+     */
+    private ensureUnique(params: Oas30Parameter[]) {
+        if (this.hasValue(params) === false) {
+            return true;
+        }
+        const resolvedParams: Oas30Parameter[] = [];
+        params.forEach(param => {
+            const resolutionResult = this.resolveParameter(param);
+            if (typeof resolutionResult === 'string') {
+                // No need to report this error - already taken care of by reference checks elsewhere.
+                // this.reportIf("PAR-3-019", true, param, "$ref", `Parameter reference ${param.$ref} points to non-existant reference.`);
+                return false;
+            }
+            resolvedParams.push(resolutionResult);
+        });
+        const paramsKey = {};
+        resolvedParams.forEach(param => {
+            const key = `${param.in}-${param.name}`;
+            if (paramsKey[key] === undefined) {
+                paramsKey[key] = {
+                    count: 1,
+                    in: param.in,
+                    name: param.name
+                };
+            }
+            else {
+                paramsKey[key]['count'] = paramsKey[key]['count'] + 1;
+            }
+        });
+        let success: boolean = true;
+        for(let key in paramsKey) {
+            if (paramsKey[key]['count'] > 1) {
+                const reportableParam = params.filter(param => {
+                    return param.in === paramsKey[key]['in'] && param.name === paramsKey[key]['name'];
+                })[0];
+                this.reportIf("PAR-3-001", true, reportableParam, "in", `Duplicate parameter named '${paramsKey[key]['name']}' and in '${paramsKey[key]['in']}' found (parameters must be unique by name and location).`);
```

Comment:
I think we should report the problem on **all** the param nodes, not just the first one.  So maybe just do a `forEach` off of the `filter`.

Reply:
Done. Updated test cases to expected errors for all parameters and not just the first.